src/integrationTest/kotlin/uk/gov/justice/digital/hmpps/educationandworkplanapi/app/resource/CreateGoalTest.kt

```diff
@@ -151,6 +153,15 @@ class CreateGoalTest : IntegrationTestBase() {
       .hasTargetDateRange(TargetDateRangeEntity.ZERO_TO_THREE_MONTHS)
       .hasStatus(StepStatus.NOT_STARTED)
       .wasCreatedBy(dpsUsername)
+
+    val expectedEventCustomDimensions = mapOf(
+      "status" to "ACTIVE",
+      "stepCount" to "1",
+      "reference" to goal.reference.toString(),
+    )
+    await.untilAsserted {
```

Comment:
`await` is not strictly necessary here as its currently sync; but this does allow me to change the implementation to an async approach and not have to worry about the integration tests